src/components/Error/ErrorBoundary.tsx

```diff
@@ -53,30 +53,30 @@ export default class ErrorBoundary extends Component<PropsWithChildren<ErrorBoun
     this.props.onError?.(error, errorInfo)
   }
 
-  renderErrorDialog(error: Error) {
+  renderErrorView(error: Error) {
     const header = error instanceof WidgetError ? error.header : DEFAULT_ERROR_HEADER
-    const action = error instanceof WidgetError ? error.action : DEFAULT_ERROR_ACTION
     return (
-      <Dialog color="dialog" forceContain>
-        <ErrorDialog
-          message={header}
-          error={error}
-          action={action}
-          onClick={
-            error instanceof WidgetError && error.dismissable
-              ? () => {
-                  this.setState({ error: undefined })
-                }
-              : () => window.location.reload()
-          }
-        />
-      </Dialog>
+      <ErrorView
+        message={header}
+        error={error}
+        action={t`Get support`}
+        onDismiss={
+          error instanceof WidgetError && error.dismissable
+            ? () => {
+                this.setState({ error: undefined })
```

Comment:
is there a reason behind why we're using a mix of class and functional react components in widget repo?

Reply:
yes - the ErrorBoundary needs to be a class component. this should be the only one afaik.

> Only class components can be error boundaries. In practice, most of the time you’ll want to declare an error boundary component once and use it throughout your application.

https://reactjs.org/docs/error-boundaries.html#introducing-error-boundaries